src/main/java/com/carrotcreative/recyclercore/widget/ProgressRecyclerViewLayout.java

```diff
@@ -112,4 +141,74 @@ public void scrollRecyclerViewToBottom()
             mRecyclerView.getLayoutManager().scrollToPosition(mRecyclerView.getAdapter().getItemCount()-1);
         }
     }
+
+    // =================================== //
+    // ========== Empty States =========== //
+    // =================================== //
+
+    /**
+     *
+     * @param emptyStateView The view that will be shown then the item count in the adapter is 0
+     */
+    public void setEmptyStateView(View emptyStateView)
+    {
+        /**
+         * If there is already an emplty state view, replace it.
+         */
+        if(mEmptyStateView != null)
+        {
+            mEmptyStateContainer.removeView(mEmptyStateView);
+        }
+        mEmptyStateView = emptyStateView;
+        mEmptyStateContainer.addView(emptyStateView);
+    }
+
+    /**
+     *
+     * @param visible true if you want to show the empty state, false other wise.
+     *                This will show the empty state only if the item count in adapter is 0.
+     *                If you want to skip item count, use #forceShowEmptyState()
+     */
+    public void setEmptyStateEnabled(boolean visible)
```

Comment:
Actually now that I think about it, I am.  Could we change `mEmptyStateVisible`, to `mEmptyStateEnabled`?